samples/snippets.py

```diff
@@ -0,0 +1,479 @@
+# Copyright 2021 Google LLC
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file or at
+# https://developers.google.com/open-source/licenses/bs
+
+"""This application demonstrates how to do basic operations with Cloud
+Spanner database.
+For more information, see the README.md under /python-spanner-sqlalchemy.
+"""
+
+from sqlalchemy import (
+    Column,
+    create_engine,
+    Index,
+    Integer,
+    inspect,
+    MetaData,
+    String,
+    Table,
+)
+from samples.conftest import insert_data
+
+DATA_1 = [
+    {"user_id": 1, "user_name": "GEH"},
+    {"user_id": 2, "user_name": "DEF"},
+    {"user_id": 3, "user_name": "HIJ"},
+    {"user_id": 4, "user_name": "ABC"},
+]
+
+DATA_2 = [
+    {"user_id": 1, "user_name": "abcdefg"},
+    {"user_id": 2, "user_name": "ab/cdefg"},
+    {"user_id": 3, "user_name": "ab%cdefg"},
+    {"user_id": 4, "user_name": "ab_cdefg"},
+    {"user_id": 5, "user_name": "abcde/fg"},
+    {"user_id": 6, "user_name": "abcde%fg"},
+]
+
+
+# [START sqlalchemy_spanner_autocommit_on]
+def enable_autocommit_mode(url):
+    """Enables autocommit mode."""
+    conn = create_engine(url).connect()
+    level = conn.get_isolation_level()
+
+    print("Connection autocommit default mode is {}".format(level))
+    print(
+        "Spanner DBAPI default autocommit mode is {}".format(
+            conn.connection.connection.autocommit
+        )
+    )
+
+    conn.execution_options(isolation_level="AUTOCOMMIT")
+    print("Connection autocommit mode is {}".format(level))
+    print(
+        "Spanner DBAPI autocommit mode is {}".format(
+            conn.connection.connection.autocommit
+        )
+    )
+    return conn
+
+
+# [END sqlalchemy_spanner_autocommit_on]
+
+
+# [START sqlalchemy_spanner_create_table]
+def create_table(url, random_table_id):
+    """Create the table."""
+    engine = create_engine(url)
+    metadata = MetaData(bind=engine)
+
+    table = Table(
+        random_table_id,
+        metadata,
+        Column("user_id", Integer, primary_key=True),
+        Column("user_name", String(16), nullable=False),
+    )
+    table.create()
+
+    print("Table {} is created successfully".format(table.name))
+    return table
+
+
+# [END sqlalchemy_spanner_create_table]
+
+
+# [START sqlalchemy_spanner_drop_table]
+def drop_table(url, random_table_id):
+    """Drop the table."""
+    engine = create_engine(url)
+    metadata = MetaData(bind=engine)
+
+    table = Table(
+        random_table_id,
+        metadata,
+        Column("user_id", Integer, primary_key=True),
+        Column("user_name", String(16), nullable=False),
+    )
+
+    table.create()
```

Comment:
This sample is for dropping a table. There is no reason to include the table creation here. That should be done in the test.

So the test should create the table and then run the sample.